File: conteudo/05-javascript/js-estrutura-de-dados/01-arrays/exemplos/palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool ehPalindromo(char *str);

int main()
{
    printf("%d\n", ehPalindromo("arara"));
    return 0;
}

bool ehPalindromo(char *str)
{
    int i_rev;

    for (int i = 0; i < strlen(str); i++)
    {
        i_rev = strlen(str) - 1 - i;
        if (str[i] != str[i_rev])
        {
            return false;
        }
    }

    return true;
}

File: conteudo/05-javascript/js-estrutura-de-dados/01-arrays/exemplos/palindromo.js

function ehPalindromo(str) {
  for (let i = 0; i < str.length / 2; i++) {
    var i_rev = str.length - 1 - i;
    if (str[i] != str[i_rev]) {
      return false;
    } 
  }
  return true;
}

console.log(ehPalindromo('arara'));
